### PR TITLE
84: moving convolution code to its own file

### LINKED ISSUE #84: Separaion map.js
Too much code in there

### PR DESCRIPTION
**Pull request by davidabrahams**
Moved convolution code to its own file. Once issue #108 is resolved, I think map.js will be lean enough. Fixes #84

---
This change is [Reviewable](https://reviewable.io/reviews/chauncy-crib/tagprobot/109)

### CODE REVIEW

File: test/map.spec.js [hunk 1]
```diff
@@ -4,14 +4,17 @@ import {
...
-  multiplyCorrespondingElementsAndSum,
-  convolve,
+  getTileTraversabilityInCells,
   __RewireAPI__ as MapRewireAPI,
 } from '../src/helpers/map';
+
+import { multiplyCorrespondingElementsAndSum,
+  getSubarrayFrom2dArray,
+  addBufferTo2dArray,
+  convolve } from '../src/helpers/convolve';
```
Comment (shanek21) [inline test/map.spec.js]: Shouldn't there now be a `convolve.spec.js` file that tests the functions in `convolve.js`?

Comment (davidabrahams) [inline test/map.spec.js]: yes. will address

**Comment by altodyte** [comment]
Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by altodyte** [comment]
:dango:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-:-Kr8apKjgmXJHy154yzl:bcbqss6)*

**Comment by billmwong** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[test/map.spec.js, line 8 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8aiG5uhzUhD7-1fX5:-Kr8aiG5uhzUhD7-1fX6:b-1wye8j) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L8)):*
> ```JavaScript
>   fillGridWithSubgrid,
>   getMapTraversabilityInCells,
>   getTileTraversabilityInCells,
> ```

Is there a reason you moved this below

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by billmwong** [comment]
*[test/map.spec.js, line 12 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8bLQs785qlwL-VkJF:-Kr8bLQs785qlwL-VkJG:burq4uq) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L12)):*
> ```JavaScript
> } from '../src/helpers/map';
> ```

why these empty lines

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by davidabrahams** [comment]
Review status: 3 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/map.spec.js, line 8 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8aiG5uhzUhD7-1fX5:-Kr8co631Ajek-5UbqeA:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L8)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Is there a reason you moved this below
</blockquote></details>

Done.

---

*[test/map.spec.js, line 12 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8bLQs785qlwL-VkJF:-Kr8cjyY8Y8gjHBJUEqP:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L12)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

why these empty lines
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by billmwong** [comment]
Review status: 3 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/map.spec.js, line 12 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8bLQs785qlwL-VkJF:-Kr8d0ITqD8CJa6rRIHO:b-9qg1u4) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L12)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Done.
</blockquote></details>

not done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by billmwong** [comment]
*[test/map.spec.js, line 12 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8bLQs785qlwL-VkJF:-Kr8d8a_LWSZL0J7X4JX:b-8fvsv5) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L12)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

not done
</blockquote></details>

:-1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by shanek21** [comment]
*[test/map.spec.js, line 12 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8bLQs785qlwL-VkJF:-Kr8dCTsfLREleC8NjvX:b-7ccl31) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L12)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

:-1: 
</blockquote></details>

:exclamation: :dango: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by shanek21** [comment]
Reviewed 1 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/helpers/convolve.js, line 3 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8dVDUdfdZkBMQdPpT:-Kr8dVDUdfdZkBMQdPpU:b-tlhhg9) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/32524a670dcfe97176ee1c903f7d26d0622feb8f/src/helpers/convolve.js#L3)):*
> ```JavaScript
> import { init2dArray, fillGridWithSubgrid } from './map';
> import { assert } from '../../src/utils/asserts';
> ```

TWO NEW LINES AFTER IMPORTS PLS

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by davidabrahams** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[test/map.spec.js, line 12 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-Kr8bLQs785qlwL-VkJF:-Kr8fIFLOiTJsX_xgAE5:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/6778387f1b48788df51d4d8bf9450ba8bdf360db/test/map.spec.js#L12)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

:exclamation: :dango: 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by shanek21** [comment]
Reviewed 1 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by billmwong** [comment]
Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 1 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by altodyte** [comment]
Reviewed 1 of 1 files at r4.
Review status: 2 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by altodyte** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by billmwong** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Comment by billmwong** [comment]
:dango:

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109#-:-Kr8g0MUm_geRJS4-S-Q:bcbqss6)*

**Comment by shanek21** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/109)*

**Pull request merged**